### PR TITLE
UNR-3761/Client-Net-Ownership-Gym-Automation

### PR DESCRIPTION
**Pull request by AndreiTeodorLazar**
#### Description
This PR adds an automated test for the Client Net Ownership Gym.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/SpatialTestNetOwnership.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			if (NetOwnershipCube->HasAuthority())
+			{
+				APlayerController* PlayerController = Cast<APlayerController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+				NetOwnershipCube->SetOwner(PlayerController);
+			}
+
+			FinishStep();
+		});
+
+	// The locations where the NetOwnershipCube will be when Client 1 will send an RPC. These are specifically set to make the NetOwnershipCube's authoritative server change according to the BP_QuadrantZoningSettings.
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/SpatialTestNetOwnership.cpp:96]: This is not future proof, as per the changes I will soon push to the GDK.
These changes will make the Cube join its owner's auth server.
In order to have the authority change, you have to move the player controller as well (which can be done by moving its possessed pawn).

Comment (AndreiTeodorLazar) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/SpatialTestNetOwnership.cpp:96]: Thank you for letting me know, will change it now!

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/NetOwnershipCube.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ANetOwnershipCube::ANetOwnershipCube()
+{
+	SetReplicateMovement(true);
+
+	ReceivedRPCs = 0;
+}
+
+void ANetOwnershipCube::ServerIncreaseRPCCount_Implementation()
+{
+	ReceivedRPCs++ ;
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/NetOwnershipCube.cpp:16]:
```suggestion
	ReceivedRPCs++;
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/NetOwnershipCube.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+
+#include "NetOwnershipCube.h"
+#include "Net/UnrealNetwork.h"
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/NetOwnershipCube.cpp:5]: double empty line at top, splitting project, GDK and engine imports
```suggestion

#include "NetOwnershipCube.h"

#include "Net/UnrealNetwork.h"
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/NetOwnershipCube.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/TestActors/ReplicatedTestActorBase.h"
+#include "NetOwnershipCube.generated.h"
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/NetOwnershipCube.h:7]:
```suggestion
#include "SpatialGDKFunctionalTests/SpatialGDK/TestActors/ReplicatedTestActorBase.h"

#include "CoreMinimal.h"

#include "NetOwnershipCube.generated.h"
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/SpatialTestNetOwnership.cpp [hunk 5]
```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+
+#include "SpatialTestNetOwnership.h"
+#include "NetOwnershipCube.h"
+#include "GameFramework/PlayerController.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "Kismet/GameplayStatics.h"
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestClientNetOwnership/SpatialTestNetOwnership.cpp:8]:
```suggestion
#include "NetOwnershipCube.h"
#include "SpatialFunctionalTestFlowController.h"
#include "SpatialTestNetOwnership.h"

#include "GameFramework/PlayerController.h"
#include "Kismet/GameplayStatics.h"
```

**Pull request merged**